File: Day - 13 (Abstract Classes)/try.c
#include <stdio.h>
#include <stdlib.h>

// Define Book class
typedef struct {
    char *title;
    char *author;
    void (*display)();
} Book;

// Define MyBook class
typedef struct {
    Book base;
    int price;
} MyBook;

// Implement display method for MyBook
void displayMyBook(MyBook *myBook) {
    printf("Title: %s\nAuthor: %s\nPrice: %d\n", myBook->base.title, myBook->base.author, myBook->price);
}

// Create a new MyBook instance
MyBook *createMyBook(char *title, char *author, int price) {
    MyBook *myBook = (MyBook *)malloc(sizeof(MyBook));
    myBook->base.title = title;
    myBook->base.author = author;
    myBook->price = price;
    myBook->base.display = (void (*)())displayMyBook;
    return myBook;
}

int main() {
    // Input from user
    char title[100], author[100];
    int price;

    printf("Enter title: ");
    fgets(title, sizeof(title), stdin);
    printf("Enter author: ");
    fgets(author, sizeof(author), stdin);
    printf("Enter price: ");
    scanf("%d", &price);

    // Create MyBook instance
    MyBook *newNovel = createMyBook(title, author, price);

    // Display information
    newNovel->base.display();

    // Free allocated memory
    free(newNovel);

    return 0;
}

File: Day - 13 (Abstract Classes)/try.java
import java.util.Scanner;

abstract class Book {
    protected String title;
    protected String author;

    public Book(String title, String author) {
        this.title = title;
        this.author = author;
    }

    public abstract void display();
}

class MyBook extends Book {
    private int price;

    public MyBook(String title, String author, int price) {
        super(title, author);
        this.price = price;
    }

    @Override
    public void display() {
        System.out.println("Title: " + title + "\nAuthor: " + author + "\nPrice: " + price);
    }
}

public class Main {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        String title = scanner.nextLine();
        String author = scanner.nextLine();
        int price = Integer.parseInt(scanner.nextLine());

        MyBook newNovel = new MyBook(title, author, price);
        newNovel.display();
    }
}
